.github/workflows/python-app.yml

```
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.13
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.13"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```

Do we need to run `ensurepip` first?  I don't remember how this works on Ubuntu systems.  This wouldn't be needed on a Fedora/RHEL system.

```suggestion
        python -m ensurepip
        python -m pip install --upgrade pip
```

---

yeah, this was based on the default GHA workflow for python app. the main thing i changed out was the calls to `pytest` and `flake8`.